apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -968,7 +968,11 @@ func (m *ModelManagerAPI) getModelInfo(ctx context.Context, tag names.ModelTag,
 	// there is no guarantee that the rest of the call will succeed.
 	// For these models we can ignore NotFound errors coming from persistence layer.
 	// However, for Alive models, these errors are genuine and cannot be ignored.
-	ignoreNotFoundError := model.Life() != state.Alive || model.MigrationMode() == state.MigrationModeImporting
+	mode, err := st.MigrationMode()
```

Comment:
@ycliuhw FYI

Reply:
I'm wondering:
why might we encounter a 'not found' error during migration with dqlite, but not with Mongo?
If a model is not alive or is in the process of migrating, it’s possible to hit a ‘not found’ error in domain services. In such cases, we should simply exit early. Currently, the info := params.ModelInfo{} data is retrieved from Mongo, but in the future, it will come from the domain layer. This means that the first fetch could legitimately return a 'not found' error.

Reply:
Above this block, we do this to get the model
```
	model, err := st.Model()
	if errors.Is(err, errors.NotFound) {
		return params.ModelInfo{}, errors.Trace(apiservererrors.ErrPerm)
```
So if the model is not found we exit at that found.
This line `mode, err := st.MigrationMode()` effectively loads the same model again under the covers. This is temporary until migration infra is moved to dqlite. To avoid the slight possibility of a model being removed during the lines of code between the 2 calls, I'll add a similar not found check here too.
The `shouldFail` check does cater for both old not found and new model not found
`isNotFound := errors.Is(thisErr, errors.NotFound) || errors.Is(thisErr, modelerrors.NotFound)`